src/main/scala/zio/prelude/NonEmptyList.scala

```diff
@@ -309,6 +338,42 @@ sealed trait NonEmptyList[+A] { self =>
   final def tails: NonEmptyList[NonEmptyList[A]] =
     unfold(self)(identity)(_.tailOption)
 
+  @tailrec
+  private def takeHelper[A](n1: Int, nel: NonEmptyList[A], taken: List[A]): List[A] =
+    if (n1 <= 0) taken
+    else
+      nel match {
+        case Cons(h, t) => takeHelper(n1 - 1, t, h :: taken)
+        case Single(h)  => h :: taken
+      }
+
+  /**
+   * Takes the first `n` elements from this `NonEmptyList` returning a `List`.
+   */
+  final def take(n: Int): List[A] =
+    takeHelper(n, self, Nil).reverse
+
+  /**
+   * Takes the last `n` elements from this `NonEmptyList` returning a `List`.
+   */
+  final def takeRight(n: Int): List[A] =
+    takeHelper(n, self.reverse, Nil)
```

Comment:
Could also be `drop(length - n)`. Then we could inline `takeHelper` into `take` like you did with `loop` below.

Reply:
Done.